commit/chainfee/outcome_test.go

```diff
@@ -398,3 +410,79 @@ func TestProcessor_Outcome(t *testing.T) {
 		})
 	}
 }
+
+func TestProcessor_OutcomeWithDifferentDecimalCalculation(t *testing.T) {
+	lggr := logger.Test(t)
+	ctx := tests.Context(t)
+	numOracles := 5
+	oneMinuteAgo := time.Now().Add(-time.Minute).UTC()
+
+	// Setup mock for home chain
+	homeChainMock := mock_home_chain.NewMockHomeChain(t)
+	homeChainMock.EXPECT().GetChainConfig(mock.Anything).
+		Return(defaultChainConfig, nil).Maybe()
+
+	// Mock the estimate provider with custom decimal calculation (1e15 instead of 1e18)
+	estimateProvider := mock_ccipocr3.NewMockEstimateProvider(t)
+	estimateProvider.EXPECT().CalculateUsdPerUnitGas(mock.Anything, mock.Anything).RunAndReturn(
+		func(sourceGasPrice *big.Int, usdPerFeeCoin *big.Int) *big.Int {
+			// Using 1e15 division instead of 1e18
+			tmp := new(big.Int).Mul(sourceGasPrice, usdPerFeeCoin)
+			return tmp.Div(tmp, big.NewInt(1e15))
+		}).Maybe()
+
+	// Create processor with custom estimate provider
+	p := &processor{
+		lggr:      lggr,
+		destChain: 1,
```

Comment:
Can we create `destChain` and `sourceChain` vars here to make the test more canonical and representative of a real situation